src/diffusers/models/unets/unet_i2vgen_xl.py

```diff
@@ -0,0 +1,679 @@
+            in_channels=block_out_channels[0],
+            num_layers=1,
+            norm_num_groups=norm_num_groups,
+        )
+
+        # image embedding
+        self.image_latents_proj_in = nn.Sequential(
+            nn.Conv2d(4, in_channels * 4, 3, padding=1),
+            nn.SiLU(),
+            nn.Conv2d(in_channels * 4, in_channels * 4, 3, stride=1, padding=1),
+            nn.SiLU(),
+            nn.Conv2d(in_channels * 4, in_channels, 3, stride=1, padding=1),
+        )
+        self.image_latents_temporal_encoder = I2VGenXLTransformerTemporalEncoder(
+            dim=in_channels,
+            num_attention_heads=2,
+            ff_inner_dim=in_channels * 4,
+            attention_head_dim=in_channels,
+            activation_fn="gelu",
+        )
+        self.image_latents_context_embedding = nn.Sequential(
+            nn.Conv2d(4, in_channels * 8, 3, padding=1),
+            nn.SiLU(),
+            nn.AdaptiveAvgPool2d((32, 32)),
+            nn.Conv2d(in_channels * 8, in_channels * 16, 3, stride=2, padding=1),
+            nn.SiLU(),
+            nn.Conv2d(in_channels * 16, cross_attention_dim, 3, stride=2, padding=1),
+        )
+
+        # other embeddings -- time, context, fps, etc.
+        time_embed_dim = block_out_channels[0] * 4
+        self.time_proj = Timesteps(block_out_channels[0], True, 0)
+        timestep_input_dim = block_out_channels[0]
+
+        self.time_embedding = TimestepEmbedding(timestep_input_dim, time_embed_dim, act_fn="silu")
+        self.context_embedding = nn.Sequential(
+            nn.Linear(cross_attention_dim, time_embed_dim),
+            nn.SiLU(),
+            nn.Linear(time_embed_dim, cross_attention_dim * in_channels),
+        )
+        self.fps_embedding = nn.Sequential(
+            nn.Linear(timestep_input_dim, time_embed_dim), nn.SiLU(), nn.Linear(time_embed_dim, time_embed_dim)
+        )
+
+        # blocks
+        self.down_blocks = nn.ModuleList([])
+        self.up_blocks = nn.ModuleList([])
+
+        if isinstance(num_attention_heads, int):
+            num_attention_heads = (num_attention_heads,) * len(down_block_types)
+
+        # down
+        output_channel = block_out_channels[0]
+        for i, down_block_type in enumerate(down_block_types):
+            input_channel = output_channel
+            output_channel = block_out_channels[i]
+            is_final_block = i == len(block_out_channels) - 1
+
+            down_block = get_down_block(
+                down_block_type,
+                num_layers=layers_per_block,
+                in_channels=input_channel,
+                out_channels=output_channel,
+                temb_channels=time_embed_dim,
+                add_downsample=not is_final_block,
+                resnet_eps=1e-05,
+                resnet_act_fn="silu",
+                resnet_groups=norm_num_groups,
+                cross_attention_dim=cross_attention_dim,
+                num_attention_heads=num_attention_heads[i],
```

Comment:
so I think it is a mistake here 
`get_down_block()` expect `num_attention_heads` to be  correct `num_attention_heads`

 you look at how `num_attention_heads` is handled inside `UNet3DConditionModel`: 

First, it fix the "bad "model config name as soon as it receives the argument, by assigning `attention_head_dim` 
 to `num_attention_heads` here, so at this point, the variable`num_attention_heads` is "corrected" for the rest of the code https://github.com/huggingface/diffusers/blob/fdf55b1f1c25c7e2fedcdd0dc9c6f3db005453bb/src/diffusers/models/unets/unet_3d_condition.py#L141

from there, we can just pass it around as it is. e.g. it is passed as `num_attention_heads` in `get_down_block()`https://github.com/huggingface/diffusers/blob/fdf55b1f1c25c7e2fedcdd0dc9c6f3db005453bb/src/diffusers/models/unets/unet_3d_condition.py#L211


in our case, we:
1. also have the "bad" model config, i.e. for this model `num_attention_heads` is actually `attention_head_dim`
2. However, we only "correct" it for `TransformerTemporalModel` here. so for the rest of the code, `num_attention_heads` still means `attention_head_dim` - and that's incorrect 
```python
self.transformer_in = TransformerTemporalModel(
            num_attention_heads=8,
            attention_head_dim=attention_head_dim,
            in_channels=block_out_channels[0],
            num_layers=1,
            norm_num_groups=norm_num_groups,
        )
```

Reply:
I am quite confused now. 

> First, it fix the "bad "model config name as soon as it receives the argument, by doing to num_attention_heads = attention_head_dim here, so at this point, the variablenum_attention_heads is "corrected" for the rest of the code

This is what we didn't want to do in this UNet i.e., the comments in https://github.com/huggingface/diffusers/pull/6665#discussion_r1473695124 suggested that there shouldn't be any `attention_head_dim`. We want to get rid of those corrections, made in `UNet3DConditionModel` for `num_attention_heads`. 

Do you mean we initialize another new variable called `attention_head_dim` based on what's passed to `num_attention_heads`?